### PR TITLE
[JENKINS-35515] - Fix the performance issue (regression in 2.3.0)

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Allow tweaking cache settings.
- [x] Disable user authorities handling by default

This change should solve the reported performance issues.
https://issues.jenkins-ci.org/browse/JENKINS-35515

@reviewbybees @jenkinsci/code-reviewers @bkmeneguello

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/rolestrategy/Settings.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    /**
+     * Enabling processing of User Authorities.
+     * Alters the behavior of 
+     * {@link RoleMap#hasPermission(java.lang.String, hudson.security.Permission, com.synopsys.arc.jenkins.plugins.rolestrategy.RoleType, hudson.security.AccessControlled)}.
+     * Since 2.3.0 this value was {@code true}, but it has been switched due to the performance reasons.
+     * The behavior can be reverted (even dynamically via System Groovy Script).
+     * @since 2.3.1
+     */
+    @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", justification = "We want to be it modifyable on the flight")
+    public static boolean THREAT_USER_AUTHORITIES_AS_ROLES =
```
Comment (batmat) [inline src/main/java/org/jenkinsci/plugins/rolestrategy/Settings.java]: @oleg-nenashev Didn't you mean _treat_ instead _threat_?

Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/rolestrategy/Settings.java]: yep :(

**Comment by sschuberth** [comment]
Nit: The commit message say "deefault" instead of "default", and also the title is a bit long and should be split into body text.

**Comment by oleg-nenashev** [comment]
@sschuberth It's not a problem, because the PR will be squashed. So the final commit message will be similar to the PR title

File: src/test/java/org/jenkinsci/plugins/rolestrategy/UserAuthoritiesAsRolesTest.java [hunk 2]
```diff
@@ -24,6 +25,16 @@
 
     @Rule public JenkinsRule j = new JenkinsRule();
 
+    @Before
+    public void enableUserAuthorities() {
+        Settings.TREAT_USER_AUTHORITIES_AS_ROLES = true;
+    }
+    
+    @Before
```
Comment (oleg-nenashev) [inline src/test/java/org/jenkinsci/plugins/rolestrategy/UserAuthoritiesAsRolesTest.java]: Typo: Should be "after"

**Comment by varmenise** [comment]
LGTM 🐝

**Comment by oleg-nenashev** [comment]
@reviewbybees done
@sschuberth Are you fine with the explanation?

**Comment by sschuberth** [comment]
@oleg-nenashev Yes, if the PR will be squash the commit message obviously does not need to be fixed. However, I personally prefer to review "the real thing" and not squash but just merge that later.

**Comment by oleg-nenashev** [comment]
I've switched to squashing after some backporting fun.
I used to do squashes before the final merge, but now GitHub saves some time by doing squash and rebase at the same time

**Pull request merged**